s3transfer/download.py

```diff
@@ -153,6 +178,55 @@ def get_final_io_task(self):
             transfer_coordinator=self._transfer_coordinator)
 
 
+
+class DownloadNonSeekableOutputManager(DownloadOutputManager):
+    def __init__(self, osutil, transfer_coordinator, io_executor,
+                 defer_queue=None):
+        super(DownloadNonSeekableOutputManager, self).__init__(
+            osutil, transfer_coordinator, io_executor)
+        if defer_queue is None:
+            defer_queue = DeferQueue()
+        self._defer_queue = defer_queue
+        self._io_submit_lock = threading.Lock()
+
+    @classmethod
+    def is_compatible(cls, download_target):
+        # We're lying a bit here.  We actually *are* compatible
+        # with seekable fileobjs, we'll still queue the IO writes
+        # in order.  Technically works, but this is not ideal behavior when you
+        # could just use DownloadSeekableOutputManager.
```

Comment:
Yeah so the way I consider ``is_compatible`` is simply if the manager could work with the fileobj given (it does not necessarily mean it is efficient). And in the end it is up to the SubmissionTask to determine, based on its own order of preference and what managers are compatible with the fileobj, what manager to use. So for your logic you would probably want to check if it is writeable. Jordon is doing a similar thing in the nonseekable case.